Data/Chimera.hs

```diff
@@ -239,13 +265,41 @@ tabulateFixM f = result
         fixF k
           | k < int2word ii
           = flip index k <$> result
-          | k < int2word ii `shiftL` 1
+          | k <= int2word ii `shiftL` 1 - 1
           = (`V.unsafeIndex` (word2int k - ii)) <$> subResultBoxed
           | otherwise
           = f fixF k
 
 {-# SPECIALIZE tabulateFixM :: G.Vector v a => ((Word -> Identity a) -> Word -> Identity a) -> Identity (Chimera v a) #-}
 
+-- | Monadic version of 'tabulateFix''.
```

Comment:
I wonder if we could deduplicate some code here. Maybe define an internal common function `tabulateFixM_`, choosing between implementations by a flag, passed as an argument?